### PR TITLE
Implementation of CLR Retraining

### PR DESCRIPTION
**Pull request by emiliamos**
- Use of CLR to retrain the Network
- prints out accuracy after L1 pruning and retraining

Issue: #35

### CODE REVIEW

File: CLR_Retraining.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import torch.nn.utils.prune as prune
+from torch.utils.data.sampler import SubsetRandomSampler
+from torch.utils.data import DataLoader
+import numpy as np
+
+# Load the GoogleNet model with auxiliary layers enabled
+model = torch.hub.load('pytorch/vision:v0.10.0', 'googlenet', weights='GoogLeNet_Weights.DEFAULT', aux_logits=True)
+model.eval()
+
+# URL for the imagenet_classes.txt file
```
Comment (leaszcz) [inline CLR_Retraining.py:18]: you can remove all unnecessary datasets and example images as they are no longer needed and choose the right dataset for validation and training

File: CLR_Retraining.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+amounts = [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
+
+# Create a multidimensional array that can store all the values
+results_global_unstructured_l1 = np.zeros(len(amounts))
+
+# Function to apply CLR
+def train_model_with_clr(model, train_loader, criterion, base_lr, max_lr, steps_per_epoch, epochs):
+    optimizer = optim.SGD(model.parameters(), lr=base_lr, momentum=0.9)
+    scheduler = torch.optim.lr_scheduler.CyclicLR(optimizer, base_lr=base_lr, max_lr=max_lr, step_size_up=max(1, steps_per_epoch//2), mode='triangular')
+
```
Comment (leaszcz) [inline CLR_Retraining.py:82]: just for informational purposes, maybe printing the current lr for each epoch would be interesting to check if the cycle from base to max and back to base works as it should

**Review by leaszcz — CHANGES_REQUESTED** [review]
Comments are in the code! Feel free to ask questions if something is unclear! :)
